pkg/azuredisk/controllerserver.go

```diff
@@ -216,7 +216,23 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 		return nil, err
 	}
 
-	selectedAvailabilityZone := pickAvailabilityZone(req.GetAccessibilityRequirements(), d.cloud.Location)
+	diskZone := pickAvailabilityZone(req.GetAccessibilityRequirements(), d.cloud.Location)
+	accessibleTopology := []*csi.Topology{
+		{
+			Segments: map[string]string{topologyKey: diskZone},
+		},
+	}
+	if skuName == compute.StandardSSDZRS || skuName == compute.PremiumZRS {
+		klog.V(2).Infof("diskZone(%s) is reset as empty since disk(%s) is ZRS(%s)", diskZone, diskName, skuName)
+		diskZone = ""
+		accessibleTopology = []*csi.Topology{}
+		for i := 1; i <= 3; i++ {
+			topology := &csi.Topology{
+				Segments: map[string]string{topologyKey: fmt.Sprintf("%s-%d", d.cloud.Location, i)},
```

Comment:
skip the `diskZone` which has already been added in L222?

Reply:
the logic is a little confusing which is still correct, let me make it more clear.

Reply:
fixed